docs/components/accordion.md

```diff
@@ -0,0 +1,195 @@
+    margin-right: 10px;
+    width: 16px;
+  }
+
+  .arrow-down:after {
+    content: "\f078";
+    font-family: "FontAwesome";
+    display: inline-block;
+    font-weight: 900;
+    height: 16px;
+    margin-right: 10px;
+    transform-origin: center;
+    width: 16px;
+  }
+}
+```
+
+## JS 
+
+_The javascript can be found in the `accordion.ts` file._ 
+
+The first line is a query selector to select all accordion widgets on the page.
+
+```js
+const accordionWidgets: NodeListOf<HTMLElement> = document.querySelectorAll("accordion-widget");
+
+//...more code here
+```
+
+The next item in the file is the `validateAccordion()` function which reviews each accordion widget and ensures it has all the necessary HTML elements.
+
+```js
+const validateAccordion = (accordionWidget: HTMLElement) => {
+
+  // Check if there are any accordion items inside the accordion widget
+  const accordionItems = accordionWidget.querySelectorAll('accordion-item');
+
+  // If there are no accordion items, log a warning
+  if (accordionItems.length === 0) {
+    console.warn('Document Error: no accordion items found inside the accordion widget.');
+    return;
+  }
+
+  // Check if there are any headings and content inside each accordion item
+  accordionItems.forEach((item, index) => {
+    const heading = item.querySelector('h3, h4');
+    const content = item.querySelector('accordion-content');
+
+    // If there are no heading for the accordion item, log a warning
+    if (!heading) {
+      console.warn(`Document Error: no heading found for accordion item ${index + 1}.`);
+      return;
+    }
+
+    // If there are no content for the accordion item, log a warning
+    if (!content) {
+      console.warn(`No content found for accordion item ${index + 1}.`);
+      return;
+    }
+  })
+
+}
+```
+
+After the validation, we define the main function that is going to add all the necessary attributes to the HTML elements. This function is called at the bottom of the file if any accordion widgets are found on the page.
+
+```js
+// Handle accordions
+const handleAccordions = () => {
+
+  // Iterate over each accordion widget
+  accordionWidgets.forEach((accordionWidget) => {
+
+    // Validate the accordion widget
+    validateAccordion(accordionWidget);
+
+    const accordionItems: NodeListOf<HTMLElement> = accordionWidget.querySelectorAll("accordion-item");
+
+    // Iterate over each accordion item
+    if (accordionItems.length > 0) {
+      accordionItems.forEach((accordionItem) => {
+
+        const accordionHeading: HTMLHeadingElement | null = accordionItem.querySelector("accordion-item>h3, accordion-item>h4");
+
+        // Add classes and attributes to the accordion heading and content
+        if (accordionHeading) {
+          accordionHeading.setAttribute("tabindex", "0");
+          accordionHeading.classList.add("toggle-btn");
+          accordionHeading.classList.add("accordion-title");
+          accordionHeading.classList.add("arrow-down");
```

Comment:
I'm not seeing the arrow change from right to bottom. What are your thoughts on adding the functionality now?

Reply:
Yeah, I've added the needed ts code.